### PR TITLE
feat: navigation bar

### PR DESCRIPTION
**Pull request by ndt080**

### CODE REVIEW

File: src/app/(main)/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import './layout.scss';
+
+import { StorageKey } from '@core/values';
+import { NavbarSize } from '@features/navbar';
+import Navbar from '@features/navbar/components/Navbar/Navbar';
+import { cookies } from 'next/headers';
+import type { PropsWithChildren } from 'react';
+
+export default function MainLayout({ children }: PropsWithChildren) {
+  const getNavbarSize = () => {
```
Comment (kirillkurko) [inline src/app/(main)/layout.tsx]: What's the reason for having the function here? Cannot we just do:
```typescript
// In your component
const navbarSize = cookies().get(StorageKey.NavbarSize)?.value as NavbarSize;

// Usage
return (
  <Navbar defaultSize={navbarSize} />
);
```

File: src/app/layout.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@ export const metadata: Metadata = {
   },
 };
 
-export default function RootLayout({ children }: PropsWithChildren) {
+export default function RootLayout({ children }: PropsWithChildren<any>) {
```
Comment (kirillkurko) [inline src/app/layout.tsx]: What's the reason for having `any` here? And cannot we go with more concrete type?

File: src/core/values/routes.ts [hunk 3]
```diff
@@ -4,4 +4,14 @@ export enum Routes {
   SignUp = '/sign-up',
   ResetPassword = '/reset-password',
   Confirmation = '/confirm',
-}
\ No newline at end of file
+  Catalog = '/catalog',
+  Profile = '/profile',
+}
```
Comment (kirillkurko) [inline src/core/values/routes.ts:9]: Enum is a set of possible options variable can be equal to, so for better semantics it's worth using singular form:
```typescript
enum Route {
  Home = '/',
  SignIn = '/sign-in',
  SignUp = '/sign-up',
  ResetPassword = '/reset-password',
  Confirmation = '/confirm',
} 
```

In this concrete case of `Route` enum means that you might have some `route` variable that must be equal to one of the members of the `Route` enum

File: src/core/values/routes.ts [hunk 4]
```diff
@@ -4,4 +4,14 @@ export enum Routes {
...
+
+export const RoutesList = [
```
Comment (kirillkurko) [inline src/core/values/routes.ts]: Cannot we get rid of the need of manually adding new routes to the `RouteList` by using `Object.values`:
```typescript
export const RoutesList = Object.values(route)
```

File: src/features/navbar/components/Navbar/Navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+interface Props {
+  defaultSize?: NavbarSize;
+}
+
+function Navbar({ defaultSize }: Props) {
+  const router = useRouter();
+  const { navbarSize, setNavbarSize } = useNavbarSize(defaultSize);
+
+  const authorized = useAuthStore((state) => state.authorized);
+  const [isOpen, setVisibility] = useState(false);
```
Comment (kirillkurko) [inline src/features/navbar/components/Navbar/Navbar.tsx]: For useState it's common to follow the naming pattern:
```typescript
const [state, setState] = useState(false);
```

So in this case the function should be renamed to `setIsOpen`

File: src/features/navbar/components/Navbar/Navbar.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+  const isShort = () => navbarSize === NavbarSize.SMALL;
```
Comment (kirillkurko) [inline src/features/navbar/components/Navbar/Navbar.tsx]: No need to use functions for such things, variable is perfectly fine: 
```typescript
const isShort = navbarSize === NavbarSize.SMALL;
```

File: src/features/navbar/components/NavbarLogo/NavbarLogo.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import React from 'react';
+
+interface Props {
+  text?: string;
+  href?: string;
+  variant?: 'text' | 'image';
+}
+
+function NavbarLogo({ text = '', href = '#', variant = 'text' }: Props) {
+  const isText = () => variant === 'text';
```
Comment (kirillkurko) [inline src/features/navbar/components/NavbarLogo/NavbarLogo.tsx]: No need to use function here

**Review by kirillkurko — APPROVED** [review]
That's a good one - there are just a few minor non-blocking comments from me. 

I'm approving this one - feel free to merge it once you address the comments 🚢

File: src/app/layout.tsx (hunk 2 above)
Comment (darkguin) [reply src/app/layout.tsx]: I forgot to remove it after tests with cookies

File: src/core/values/routes.ts (hunk 4 above)
Comment (darkguin) [reply src/core/values/routes.ts]: @kirillkurko If memory does not change, then enum had problems with this when both keys and values were returned to Object.values. Perhaps this has already been corrected in new versions, you need to check

File: src/features/navbar/components/Navbar/Navbar.tsx (hunk 6 above)
Comment (darkguin) [reply src/features/navbar/components/Navbar/Navbar.tsx]: will the isShort be updated every time navbarSize is updated? The idea was to make an alias and take it out of the template

File: src/features/navbar/components/NavbarLogo/NavbarLogo.tsx (hunk 7 above)
Comment (darkguin) [reply src/features/navbar/components/NavbarLogo/NavbarLogo.tsx]: a question similar to the comment above

File: src/core/values/routes.ts (hunk 4 above)
Comment (darkguin) [reply src/core/values/routes.ts]: Yes, it already works correctly

File: src/features/navbar/components/Navbar/Navbar.tsx (hunk 6 above)
Comment (kirillkurko) [reply src/features/navbar/components/Navbar/Navbar.tsx]: Actually `isShort` value will be evaluated not only when `navbarSize` changes, but every time the component is rendered. For heave computations you might want to optimize it using `useMemo` hook, but here it's perfectly fine

File: src/core/values/routes.ts (hunk 4 above)
Comment (kirillkurko) [reply src/core/values/routes.ts]: Yep, it works fine now (actually I double-checked it before leaving my first comment)

File: src/features/navbar/components/Navbar/Navbar.tsx (hunk 6 above)
Comment (ndt080) [reply src/features/navbar/components/Navbar/Navbar.tsx]: To summarize. `const isShort = navbarSize === NavbarSize.SMALL` will be updated:
- when is the re-render
- when are the props changed
- when navbarSize is updated
?

Do I understand correctly that we only do callbacks when we want to optimize using useMemo/useCallback?

Comment (ndt080) [reply src/features/navbar/components/Navbar/Navbar.tsx]: If you have a good article about the change detection mechanism in react, I would read it :)

Comment (kirillkurko) [reply src/features/navbar/components/Navbar/Navbar.tsx]: These are two great articles to better understand re-rendering in React:
- https://www.developerway.com/posts/react-re-renders-guide
- https://www.joshwcomeau.com/react/why-react-re-renders/

And you might also want to take a look useMemo/useCallback usages:
- https://www.developerway.com/posts/how-to-use-memo-use-callback
- https://www.joshwcomeau.com/react/usememo-and-usecallback/

Comment (kirillkurko) [reply src/features/navbar/components/Navbar/Navbar.tsx]: We do callback when we really need to use callback (e.g. to pass it down the tree and call from the children component or if it involves some computation that you want to keep all together)

**Pull request merged**